Cython/Compiler/ExprNodes.py

```diff
@@ -9899,6 +9899,8 @@ def generate_yield_code(self, code):
             save_cname = code.funcstate.closure_temps.allocate_temp(type)
             saved.append((cname, save_cname, type))
             code.put_xgiveref(cname, type)
+            if type.is_cpp_class:
+                cname = "__PYX_STD_MOVE_IF_SUPPORTED(%s)" % cname
```

Comment:
These two cases are exclusive.
```suggestion
            if type.is_cpp_class:
                cname = "__PYX_STD_MOVE_IF_SUPPORTED(%s)" % cname
            else:
                code.put_xgiveref(cname, type)
```